cuebot/src/main/resources/conf/ddl/postgres/migrations/V3__Increase_show_length.sql

```diff
@@ -0,0 +1,7 @@
+-- Increase the length of show names
+
+-- Due to the name being used in other views we need to increase the length
+-- by altering the pg_attributes, otherwise we need to drop the depedent views
+UPDATE pg_attribute SET atttypmod = 512+4 WHERE attrelid IN ('show'::regclass, 'v_history_job'::regclass, 'v_history_layer'::regclass) AND attname = 'str_name';
```

Comment:
I believe for `v_history_job` and `v_history_layer` you want to modify the `str_show_name` column.

My preference here would be to drop and recreate the view, any reason to prefer updating the attributes as above?

Reply:
Good spot, dropping and re-creating would duplicate the code for the views in the migration which from my reading doesn't seem necessary if using this method.

Reply:
If you're happier with duplicating the view code in the migration I don't mind changing it.

Reply:
There is a risk when altering the catalogs directly since it essentially removes the safety rails, but I do agree the duplication of the view creation is annoying. @bcipriano Any perference?

Reply:
Hm, it's a tough call but I think the `pg_attribute` approach feels pretty obscure and a bit hard to understand (it took me a few minutes to parse the query and understand what was going on). And I agree it feels dangerous, easy to mistype a field which could easily lead to data corruption.

OTOH it is annoying to duplicate all those view schemas.

I think it's worth being cautious here because this kind of operation will certainly need to happen in the future so whatever we decide here will set a precedent for future migrations.

Does anyone know of some cleaner way to dump a view schema then restore it?

If we can't find some third way, I'd vote for just duplicating the view code. It feels safer and more explicit IMO.